packages/graphql/src/translate/cypher-builder/build-merge-statement.ts

```diff
@@ -90,6 +91,15 @@ export function buildMergeStatement({
             });
     }
 
+    if (sourceNode.parameters) {
+        parameters = sourceNode.node?.constrainableFields.reduce((params, field) => {
+            if (Object.keys(sourceNode.parameters!).includes(field.fieldName)) {
+                params[field.dbPropertyName || field.fieldName] = sourceNode.parameters![field.fieldName];
+            }
+            return params;
+        }, {} as Record<string, any>);
+    }
```

Comment:
@darrellwarde I think that can be solved with an intermediate variable. Because it is a parameter checked outside reduce, ts is not that clever as to guarantee it hasn't been changed in the reduce.

```suggestion
    const nodeParameters = sourceNode.parameters;
    if (nodeParameters) {
        parameters = sourceNode.node?.constrainableFields.reduce((params, field) => {
            if (Object.keys(nodeParameters).includes(field.fieldName)) {
                params[field.dbPropertyName || field.fieldName] = nodeParameters[field.fieldName];
            }
            return params;
        }, {} as Record<string, any>);
    }
```